drivers/media/i2c/imx708.c

```diff
@@ -1835,11 +1890,15 @@ static void imx708_free_controls(struct imx708 *imx708)
 
 static int imx708_check_hwcfg(struct device *dev)
 {
+	struct i2c_client *client = to_i2c_client(dev);
+	struct v4l2_subdev *sd = i2c_get_clientdata(client);
+	struct imx708 *imx708 = to_imx708(sd);
```

Comment:
That's a heck of a dance. Just pass ```struct imx708 *imx708``` in from ```imx708_probe```.

Reply:
I can just pass in struct imx708 * into the function to make it cleaner.